contracts/engagement/src/contract.rs

```diff
@@ -91,18 +91,30 @@ impl EngagementContract {
             Ok(esc) => esc,
             Err(err) => return Err(err),
         };
+
+        if escrow.dispute_flag == false {
+            return Err("Escrow has been opened for dispute resolution");
+        }
     
-        let half_price_in_micro_usdc = (escrow.amount as i128) / 2;
         let usdc_client = TokenClient::new(&e, &usdc_contract);
-    
+
         let signer_balance = usdc_client.balance(&signer);
-        if signer_balance < half_price_in_micro_usdc {
+        
+        if escrow.balance > escrow.amount {
```

Comment:
The escrow balance should be obtained in the same way you did with the “signer_balance” variable, not from the “balance” property of the escrow.
In fact, an apology, since the “balance” property should not be part of the escrow properties. After the ODHACK we will fix it.

Reply:
got it! thank you

Reply:
Try to avoid using escrow.balance entirely, as it will be removed in future versions. Instead, you can use usdc_client.balance(contract_address). The contract_address variable is already declared on line 115, so you can leverage that and completely eliminate the dependency on escrow.balance.

Reply:
Done! ✅